core/src/main/scala/kafka/server/BrokerToControllerRequestManager.scala

```diff
@@ -125,14 +125,18 @@ abstract class AbstractBrokerToControllerRequestManager[Item <: BrokerToControll
   private def sendRequest(itemsToSend: Seq[Item]): Unit = {
     val brokerEpoch = brokerEpochSupplier.apply()
     val request = buildRequest(itemsToSend, brokerEpoch)
-    debug(s"Sending to controller $request")
+    info(s"Sending to controller ${request.getClass.getName} of ~${request.toString.length} bytes (${itemsToSend.length} items)")
+    debug(s"Full request: $request")
 
     // We will not timeout the request, instead letting it retry indefinitely
     // until a response is received.
     controllerChannelManager.sendRequest(request,
       new ControllerRequestCompletionHandler {
         override def onComplete(response: ClientResponse): Unit = {
-          debug(s"Received response $response")
+          // several tests create ClientResponses with null requestHeaders
+          val requestType = if (response.requestHeader != null) response.requestHeader.apiKey.name else "unknown-BrokerToControllerRequest"
```

Comment:
Can you ensure "response.requestHeader.apiKey != null"? If not, add a check along with "response.requestHeader != null"

Reply:
I can do a followup, but I considered that possibility and ruled it out, since what kind of request would the broker make to the controller that doesn't involve an API?  It's kind of a contradiction in terms.